classes/village/VillageManager.php

```diff
@@ -347,25 +347,74 @@ public static function claimSeat(System $system, User $player, string $seat_type
     }
 
     /**
-     * @return bool
+     * @return string
      */
     public static function resign(System $system, User $player): string
     {
+        $message = '';
+        $time = time();
+        $player_seat = $player->village_seat;
+        $player_seat_id = $player_seat->seat_id;
+        $player_seat_type = $player_seat->seat_type;
+        // get challenger if set
+        $challenger = null;
```

Comment:
```suggestion
        $first_challenger = null;
```
to make the intended behavior here more clear

Reply:
> to make the intended behavior here more clear

Done!